utils/scarb-ui/src/args/packages_filter.rs

```diff
@@ -99,6 +102,23 @@ impl PackagesFilter {
         Self::do_match_all::<S>(specs, self.workspace, members)
     }
 
+    /// Get the packages filter as a OsString.
+    ///
+    /// This value can be passed as `SCARB_PACKAGES_FILTER` variable to child processes.
+    pub fn build_packages_filter<S: PackagesSource>(&self, source: &S) -> Result<OsString> {
```

Comment:
I don't feel convinced this will be a good API. Mind that this is a library code, we need to think about such things.


I wonder if it wouldn't make sense to split this into two phases:

```rust
pub fn generate_for<S: PackagesSource>(source: &S) -> Self { todo!() }
pub fn to_env(&self) -> OsString { todo!() }

impl PackagesSource for &[Package] {}
```

This way, we can technically generate packages filter for an already filtered workspace without bothering who actually did the filtering. Wdyt?